cypress/cypress/integration/mentor-card.spec.ts

```diff
@@ -0,0 +1,168 @@
+Permission to use, copy, modify, and distribute this software and its documentation for educational, research and non-profit purposes, without fee, and without a written agreement is hereby granted, provided that the above copyright notice and subject to the full license file found in the root of this software deliverable. Permission to make commercial use of this software may be obtained by contacting:  USC Stevens Center for Innovation University of Southern California 1150 S. Olive Street, Suite 2300, Los Angeles, CA 90115, USA Email: accounting@stevens.usc.edu
+
+The full terms of this copyright and license should always be found in the root directory of this software deliverable as "license.txt" and if these terms are not found with this software, please contact the USC Stevens Center for the full license.
+*/
+import { cySetup, cyMockDefault } from "../support/functions";
+import clint from "../fixtures/mentor/clint_home";
+import { Status, QuestionType, UtteranceName } from "../support/types";
+
+describe("Mentor Card on My Mentor Page", () => {
+  it("mentor card contains correct data", () => {
+    cySetup(cy);
+    cyMockDefault(cy, { mentor: clint });
+    cy.visit("/");
+    cy.get("[data-cy=my-mentor-card]").contains("Clinton Anderson");
+    cy.get("[data-cy=my-mentor-card]").contains("Video Mentor");
+    cy.get("[data-cy=my-mentor-card]").contains(
+      "Title: Nuclear Electrician's Mate"
+    );
+    cy.get("[data-cy=my-mentor-card]").contains("Last Trained: Today");
+    cy.get("[data-cy=my-mentor-card]").contains("Scope: Incomplete");
+    cy.get("[data-cy=my-mentor-card]").contains(
+      "This Mentor can't be built yet."
+    );
+    cy.get("[data-cy=stage-progress]").should("exist");
+    cy.get("[data-cy=my-mentor-card]")
+      .find("[data-cy=next-stage-info]")
+      .trigger("mouseover");
+    cy.contains("This Mentor can select questions from a list");
+    cy.get("[data-cy=thumbnail-wrapper]").trigger("mouseover");
+    cy.get("[data-cy=upload-file]").should("exist");
+    cy.get("[data-cy=bash-button]").should("exist");
+  });
+
+  it("shows placeholder when no thumbnail", () => {
+    cySetup(cy);
+    const testClint = {
+      ...clint,
+      thumbnail: "",
+    };
+    cyMockDefault(cy, { mentor: testClint });
+    cy.visit("/");
+    cy.get("[data-cy=placeholder-thumbnail]").should("exist");
+  });
+
+  it("switches to new image when uploaded", () => {
+    cySetup(cy);
+    const testClint = {
+      ...clint,
+      thumbnail: "url",
+    };
+    cyMockDefault(cy, { mentor: testClint });
+    cy.visit("/");
+
+    cy.get("[data-cy=uploaded-thumbnail]").should("exist");
+  });
+
+  it("does not show toast on incomplete level", () => {
+    cySetup(cy);
+    cyMockDefault(cy, { mentor: clint });
+    cy.visit("/");
+    cy.get("[data-cy=stage-toast]").should("not.exist");
+  });
+
+  it("shows mentor scope toast on stage floor", () => {
+    cySetup(cy);
+    const testClint = {
```

Comment:
rule of thumb: always inline a variable if you can. The reason to NOT inline would be if other things need to access the variable by name; I think here they don't so...

```typescript
cyMockDefault(cy, { 
  mentor: {
    ...clint,
    // ...
  }
});